vllm/model_executor/kernels/linear/mixed_precision/gfx900_w4a16_gemv.py

```
@@ -0,0 +1,115 @@
+    shifts2d = shifts[None, :]
+
+    g0 = pid_k * GROUPS_PER_PID
+    for gi in range(0, GROUPS_PER_PID):
+        g = g0 + gi
+        in_range = g < (K // G)
+        s_ptrs = scales_ptr + g * N + offs_n
+        scales = tl.load(s_ptrs, mask=mask_n & in_range, other=0.0).to(tl.float32)
+        if HAS_ZP:
+            qz_ptrs = qz_ptr + g * (N // 8) + offs_n8
+            qzv = tl.load(qz_ptrs, mask=mask_n8 & in_range, other=0)
+            ze = tl.interleave(qzv, qzv); ze = tl.interleave(ze, ze); ze = tl.interleave(ze, ze)
```

Ruff (0.15.20)

[error] Multiple statements on one line (semicolon)

(E702)